Bottom: c939273aa2f357eaf752d05a6826b3e99c4a75ae
Top:    3f5e1d917dfaeeb9650072d2dc589ee4f755e4cd
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-21 22:27:13 +0000

Refresh of major-re-write-to-drawing-rout

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 436c06a..0b9e171 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1264,20 +1264,6 @@ hole_callback (const BoxType * b, void *cl)
 }
 
 static void
-_draw_line (LineType *line)
-{
-  gui->graphics->set_line_cap (Output.fgGC, Trace_Cap);
-  if (TEST_FLAG (THINDRAWFLAG, PCB))
-    gui->graphics->set_line_width (Output.fgGC, 0);
-  else
-    gui->graphics->set_line_width (Output.fgGC, line->Thickness);
-
-  gui->graphics->draw_line (Output.fgGC,
-		  line->Point1.X, line->Point1.Y,
-		  line->Point2.X, line->Point2.Y);
-}
-
-static void
 draw_line (LayerType *layer, LineType *line)
 {
   if (TEST_FLAG (SELECTEDFLAG | FOUNDFLAG, line))
@@ -1289,7 +1275,7 @@ draw_line (LayerType *layer, LineType *line)
     }
   else
     gui->graphics->set_color (Output.fgGC, layer->Color);
-  _draw_line (line);
+  gui->graphics->draw_pcb_line (Output.fgGC, line);
 }
 
 static int
@@ -1300,22 +1286,6 @@ line_callback (const BoxType * b, void *cl)
 }
 
 static void
-_draw_arc (ArcType *arc)
-{
-  if (!arc->Thickness)
-    return;
-
-  if (TEST_FLAG (THINDRAWFLAG, PCB))
-    gui->graphics->set_line_width (Output.fgGC, 0);
-  else
-    gui->graphics->set_line_width (Output.fgGC, arc->Thickness);
-  gui->graphics->set_line_cap (Output.fgGC, Trace_Cap);
-
-  gui->graphics->draw_arc (Output.fgGC, arc->X, arc->Y, arc->Width,
-                 arc->Height, arc->StartAngle, arc->Delta);
-}
-
-static void
 draw_arc (LayerType *layer, ArcType *arc)
 {
   if (TEST_FLAG (SELECTEDFLAG | FOUNDFLAG, arc))
@@ -1328,7 +1298,7 @@ draw_arc (LayerType *layer, ArcType *arc)
   else
     gui->graphics->set_color (Output.fgGC, layer->Color);
 
-  _draw_arc (arc);
+  gui->graphics->draw_pcb_arc (Output.fgGC, arc);
 }
 
 static int
